vivisect/__init__.py

```diff
@@ -630,13 +630,14 @@ def initWorkspaceClient(self, remotevw):
         self.server.vprint('%s connecting...' % uname)
         wsevents = self.server.exportWorkspace()
         self.importWorkspace(wsevents)
-        self._snapInAnalysisModules()
         self.server.vprint('%s connection complete!' % uname)
 
         thr = threading.Thread(target=self._clientThread)
         thr.setDaemon(True)
         thr.start()
 
+        self._snapInAnalysisModules()
```

Comment:
i was thinking about that.  
only way to be certain is checking `getMeta('Architecture')` and `getMeta('Platform')`,  but that would require that all shared workspaces implement those no matter what.  is that currently a given?  
alternately, we could wait a specified amount of time for those to be marked.  

the current race-condition involves trying to snap in analysis modules before those are set.  currently, the analysis modules only matter once the user/calling code is returned control.

Reply:
Hmmm, since it's all in the same process, we might be able to do some shenanigans with python `threading.Event` to force a sync between the threads.

And as always...test :smiley:?